crates/matrix-sdk/src/room_directory_search.rs

```diff
@@ -118,6 +118,7 @@ impl SearchState {
 pub struct RoomDirectorySearch {
     batch_size: u32,
     filter: Option<String>,
+    server: Option<OwnedServerName>,
```

Comment:
ditto, how about `via_server` or just `via` that i think we've used in other places?